### PR TITLE
RCInput: move SBUS and PPM to same pin

### PR DESCRIPTION
**Pull request by elizabethshirley**
This PR fixes the Zynq driver so PPM and SBUS can be used on the same pin.
This PR also removes the unnecessary macros in the linux board header, and cleans up the RCInput selection.

**Fix Details:**
During _process_rc_pulse() in RCInput.cpp, the process_ppmsum_pulse function only relies on the length of the signal, so input is polarity agnostic; which is why the PPM worked before the fix. 

However, SBUS signals are not polarity agnostic. Flipping the logic levels on the data received from the RC receiver seems to have fixed the issue.

### CODE REVIEW

File: libraries/AP_HAL_Linux/RCInput_ZYNQ.cpp [hunk 1]
```diff
@@ -32,10 +32,10 @@ void RCInput_ZYNQ::init()
     if (mem_fd == -1) {
         AP_HAL::panic("Unable to open /dev/mem");
     }
-    pulse_input = (volatile uint32_t*) mmap(0, 0x1000, PROT_READ|PROT_WRITE,
+
+    pulse_input = (volatile uint32_t*) mmap(NULL, 0x1000, PROT_READ|PROT_WRITE,
```
Comment (davidaroyer) [inline libraries/AP_HAL_Linux/RCInput_ZYNQ.cpp:36]: Forgive my ignorance, but what's the advantage to switching to NULL here?

File: libraries/AP_HAL_Linux/RCInput_ZYNQ.cpp [hunk 2]
```diff
@@ -45,13 +45,17 @@ void RCInput_ZYNQ::init()
 void RCInput_ZYNQ::_timer_tick()
 {
     uint32_t v;
-
```
Comment (davidaroyer) [inline libraries/AP_HAL_Linux/RCInput_ZYNQ.cpp:48]: This is super picky, but I'd rather keep this empty line here for the sake of readability.

File: libraries/AP_HAL_Linux/RCInput_ZYNQ.cpp [hunk 3]
```diff
@@ -45,13 +45,17 @@ void RCInput_ZYNQ::init()
...
     // all F's means no samples available
     while((v = *pulse_input) != 0xffffffff) {
-        // Hi bit indicates pin state, low bits denote pulse length
-        if(!(v & 0x80000000))
-            _s0_time = (v & 0x7fffffff)/TICK_PER_US;
-        else
-            _process_rc_pulse(_s0_time, (v & 0x7fffffff)/TICK_PER_US);
+        // This calculation is done everytime
```
Comment (davidaroyer) [inline libraries/AP_HAL_Linux/RCInput_ZYNQ.cpp]: I don't think this comment is really necessary. It's perfectly fine as justification for the cleanup here, just no need to add it in here.

File: libraries/AP_HAL_Linux/RCInput_ZYNQ.cpp [hunk 4]
```diff
@@ -45,13 +45,17 @@ void RCInput_ZYNQ::init()
...
+        auto temp = (v & 0x7fffffff)/TICK_PER_US;
+        // Hi bit indicates low pin state
+        // Lower 31 bits denote pulse length
+        if (v & 0x80000000) {
```
Comment (davidaroyer) [inline libraries/AP_HAL_Linux/RCInput_ZYNQ.cpp]: Just to be sure, this is the opposite of what was in the current driver, right? Can you add some explanation for this change in your original PR comment?

**Review by davidaroyer — COMMENTED** [review]
Mostly good to go; see comments.

File: libraries/AP_HAL_Linux/RCInput_ZYNQ.cpp (hunk 1 above)
Comment (elizabethshirley) [reply libraries/AP_HAL_Linux/RCInput_ZYNQ.cpp:36]: Since it's an address, both are NULL, just one is more verbose.

**Comment by davidaroyer** [comment]
Commits cleaned up, and PR successfully tested.

**Pull request merged**